cmd/opampsupervisor/e2e_test.go

```diff
@@ -273,6 +280,42 @@ func TestSupervisorStartsCollectorWithRemoteConfig(t *testing.T) {
 	}, 10*time.Second, 500*time.Millisecond, "Log never appeared in output")
 }
 
+func TestSupervisorStartsCollectorWithLocalConfigOnly(t *testing.T) {
+	connected := atomic.Bool{}
+	server := newOpAMPServer(t, defaultConnectingHandler, types.ConnectionCallbacks{
+		OnConnected: func(_ context.Context, _ types.Connection) {
+			connected.Store(true)
+		},
+	})
+
+	cfg, _, _, _ := createSimplePipelineCollectorConf(t)
+
+	collectorConfigDir := t.TempDir()
+	cfgFile, err := os.CreateTemp(collectorConfigDir, "config_*.yaml")
+	t.Cleanup(func() { cfgFile.Close() })
+	require.NoError(t, err)
+
+	_, err = cfgFile.Write(cfg.Bytes())
+	require.NoError(t, err)
+
+	storageDir := t.TempDir()
+
+	s := newSupervisor(t, "basic", map[string]string{
+		"url":          server.addr,
+		"storage_dir":  storageDir,
+		"local_config": cfgFile.Name(),
+	})
+	t.Cleanup(s.Shutdown)
+	require.NoError(t, s.Start())
+
+	waitForSupervisorConnection(server.supervisorConnected, true)
+	require.True(t, connected.Load(), "Supervisor failed to connect")
+
+	require.EventuallyWithTf(t, func(c *assert.CollectT) {
+		require.NotContains(c, getAgentLogs(t, storageDir), "error")
```

Comment:
Is there another way we can verify that the Collector has successfully connected to the Supervisor? I'm worried this check is a bit too broad.

Reply:
@evan-bradley I thought the same as I was developing it. The only alternative way I could think to check whether the Collector connected back would require me to export a method in the Supervisor struct only to be able to use it in the e2e test. If that's okay for you I am happy to go this.

Reply:
I could improve this test without adding an extra method to the `Supervisor` struct.

Now it's checking for the presence of the line `Connected to the OpAMP server`, which is there if the Collector log level is set to `debug` (which I now configure in the simple pipeline conf used by the test).

Let me know please whether this is good or if you prefer the exported method in the `Supervisor` struct.

Reply:
Fixed in [`54ee558` (#39949)](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39949/commits/54ee558c319bc502edf9793f737bb455fd80bb38)

Reply:
That seems specific enough to me, thank you.